web/src/test/java/org/springframework/security/web/authentication/logout/CompositeLogoutHandlerTests.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2002-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.web.authentication.logout;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import org.springframework.test.util.ReflectionTestUtils;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Eddú Meléndez
+ */
+public class CompositeLogoutHandlerTests {
+
+	@Rule
+	public ExpectedException exception = ExpectedException.none();
+
+	@Test
+	public void buildEmptyCompositeLogoutHandlerThrowsException() {
+		this.exception.expect(IllegalArgumentException.class);
+		this.exception.expectMessage("LogoutHandlers are required");
+		new CompositeLogoutHandler();
+	}
+
+	@Test
+	public void buildCompositeLogoutHandlerWithArray() {
+		LogoutHandler[] logoutHandlers = {new SecurityContextLogoutHandler()};
+		LogoutHandler handler = new CompositeLogoutHandler(logoutHandlers);
+		assertThat(ReflectionTestUtils.getField(handler, "logoutHandlers")).isNotNull();
+		assertThat(((List<LogoutHandler>)ReflectionTestUtils.getField(handler,
+				"logoutHandlers")).size())
+				.isEqualTo(1);
+	}
+
+	@Test
+	public void buildCompositeLogoutHandlerWithList() {
+		LogoutHandler securityContextLogoutHandler = new SecurityContextLogoutHandler();
+		List<LogoutHandler> logoutHandlers = Arrays.asList(securityContextLogoutHandler);
+		LogoutHandler handler = new CompositeLogoutHandler(logoutHandlers);
+		assertThat(ReflectionTestUtils.getField(handler, "logoutHandlers")).isNotNull();
+		assertThat(((List<LogoutHandler>)ReflectionTestUtils.getField(handler,
+				"logoutHandlers")).size())
+				.isEqualTo(1);
+	}
+
```

Comment:
Can we add a test that ensures that the logoutHandlers that are injected are invoked when CompositeLogoutHandler.logout is invoked? This would likely work well with a mock.

We should also document what happens if an Exception happens in one of the LogoutHandlers (the remaining LogoutHandlers won't be invoked) and have a test to verify the documentation is correct.